shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -345,6 +362,235 @@ class UCX(executor: BlockManagerId, rapidsConf: RapidsConf) extends AutoCloseabl
+                  s"'${rapidsConf.shuffleUcxActiveMessagesMode}' as our configured mode.")
+              logDebug(s"Handling an EAGER active message receive ${amData}")
+              val resp = UcxUtils.getByteBufferView(amData.getDataAddress, amData.getLength)
+
+              // copy the data onto a buffer we own because it is going to be reused
+              // in UCX
+              val dbb = cb.onHostMessageReceived(amData.getLength)
+              val bb = dbb.getBuffer()
+              bb.put(resp)
+              bb.rewind()
+              cb.onSuccess(am, dbb)
+
+              // we return OK telling UCX `amData` is ok to be closed, along with the eagerly
+              // received data
+              UcsConstants.STATUS.UCS_OK
+            } else {
+              // RNDV case: we get a direct buffer and UCX will fill it with data at `receive`
+              // callback
+              val resp = cb.onHostMessageReceived(amData.getLength)
+
+              val receiveAm = amData.receive(UcxUtils.getAddress(resp.getBuffer()),
+                new UcxCallback {
+                  override def onError(ucsStatus: Int, errorMsg: String): Unit = {
+                    withResource(resp) { _ =>
+                      withResource(amData) { _ =>
+                        if (ucsStatus == UCX.UCS_ERR_CANCELED) {
+                          logWarning(
+                            s"Cancelled Active Message " +
+                              s"${TransportUtils.toHex(reg.activeMessageId)}" +
+                              s" status=$ucsStatus, msg=$errorMsg")
+                          cb.onCancel(am)
+                        } else {
+                          cb.onError(am, ucsStatus, errorMsg)
+                        }
+                      }
+                    }
+                  }
+
+                  override def onSuccess(request: UcpRequest): Unit = {
+                    withResource(amData) { _ =>
+                      cb.onSuccess(am, resp)
+                    }
+                  }
+                })
+
+              cb.onMessageStarted(receiveAm)
+              UcsConstants.STATUS.UCS_INPROGRESS
+            }
+          }
+        })
+    })
+  }
+
+  // If we are not forcing RNDV (i.e. we are in auto or eager) other handling
+  // can happen when we receive an Active Message message (it can contain
+  // inline data that must be copied out in the callback).
+  private lazy val notForcingAmRndv: Boolean = {
+    !rapidsConf.shuffleUcxActiveMessagesMode
+      .equalsIgnoreCase("rndv")
+  }
+
+  private lazy val activeMessageMode: Long = {
+    rapidsConf.shuffleUcxActiveMessagesMode match {
+      case "eager" =>
+        UcpConstants.UCP_AM_SEND_FLAG_EAGER
+      case "rndv" =>
+        UcpConstants.UCP_AM_SEND_FLAG_RNDV
+      case "auto" =>
+        0L
+      case _ =>
+        throw new IllegalArgumentException(
+          s"${rapidsConf.shuffleUcxActiveMessagesMode} is an invalid Active Message mode. ")
```

Comment:
This message needs to mention `RapidsConf.SHUFFLE_UCX_ACTIVE_MESSAGES_MODE.key` so the user knows this came from an invalid setting of that config.